### PR TITLE
rcll-central: approach BS from free side when in use

### PR DESCRIPTION
**Pull request by kalrasamridhi7**

### CODE REVIEW

File: src/clips-specs/rcll-central/derived-predicates/bs-side-in-use.clp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+(defrule bs-side-in-use
+"If a BS is part of a goal's operation, assert a fact to indicate this state."
+  (declare (salience ?*SALIENCE-DOMAIN-CHECK*))
+	(not (domain-fact (name bs-side-in-use) (param-values bs $?)))
```
Comment (TarikViehmann) [inline src/clips-specs/rcll-central/derived-predicates/bs-side-in-use.clp:4]:
```suggestion
	(not (domain-fact (name bs-side-in-use) (param-values ?bs $?)))
```
and this precond needs to go after the one that grounds ths ?bs variable

File: src/clips-specs/rcll-central/derived-predicates/bs-side-in-use.clp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	(wm-fact (key domain fact mps-type args? $? ?bs $? BS $?))
```
Comment (TarikViehmann) [inline src/clips-specs/rcll-central/derived-predicates/bs-side-in-use.clp:5]:
```suggestion
	(domain-fact (name mps-type) (param-values ?bs BS))
```

File: src/clips-specs/rcll-central/derived-predicates/bs-side-in-use.clp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+               (state FORMULATED|PENDING|WAITING|RUNNING))
+	=>
+	(assert (domain-fact (name bs-side-in-use) (param-values ?bs ?bs-side)))
+)
+
+(defrule bs-side-in-use-negative
+"Retract BS in use fact if it is no longer in use."
+  (declare (salience ?*SALIENCE-DOMAIN-CHECK*))
+	?wm <- (domain-fact (name bs-side-in-use) (param-values ?bs ?bs-side))
+	(wm-fact (key domain fact mps-type args? $? ?bs $? BS $?))
```
Comment (TarikViehmann) [inline src/clips-specs/rcll-central/derived-predicates/bs-side-in-use.clp:17]:
```suggestion
	(domain-fact (name mps-type) (param-values ?bs BS))
```

**Review by TarikViehmann — CHANGES_REQUESTED** [review]
Good job, it is also tested in practice.
But have a few small comments that should be addressed.

**Comment by kalrasamridhi7** [comment]
Addressed the requested changes

**Review by TarikViehmann — APPROVED** [review]
Thank you very much!

**Pull request merged**